src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -199,17 +207,15 @@ public Activity AddBaggage(string key, string value)
         public Activity SetParentId(string parentId)
         {
             if (Parent != null)
-                throw new InvalidOperationException(
-                    $"Trying to set {nameof(ParentId)} on activity which has {nameof(Parent)}");
-
-            if (ParentId != null)
-                throw new InvalidOperationException(
-                    $"{nameof(ParentId)} is already set");
-
-            if (string.IsNullOrEmpty(parentId))
-                throw new ArgumentException($"{nameof(parentId)} must not be null or empty");
-
-            ParentId = parentId;
+                NotifyError(new InvalidOperationException($"Trying to set {nameof(ParentId)} on activity which has {nameof(Parent)}"));
```

Comment:
Since we're no longer throwing exceptions, how does a typical developer know that they're doing something wrong?  Seems like the common case will be that they end up shipping a production solution that incurs a lot of exception-related runtime costs and they'll never know it.